File: packages/dart/lib/src/territorial/classify_hazard.dart
import 'territorial_data.dart';

String classifyHazard(String text) {
  final lower = text.toLowerCase();
  for (final rule in TerritorialData.hazardRules) {
    final keywords = rule['keywords'] as List<dynamic>;
    for (final keyword in keywords) {
      if (lower.contains(keyword as String)) {
        return rule['type'] as String;
      }
    }
  }
  return TerritorialData.defaultHazard;
}

File: packages/go/classify_hazard.go
package perspatial

import "strings"

func ClassifyHazard(text string) string {
	lower := strings.ToLower(text)
	for _, rule := range hazards.Rules {
		for _, kw := range rule.Keywords {
			if strings.Contains(lower, kw) {
				return rule.Type
			}
		}
	}
	return hazards.Default
}
